docs/Developers/Profile_Bears.rst

```diff
@@ -0,0 +1,76 @@
+Profile Bears
+=============
+
+This document provides an overview of coala's Profiling Interface. The
+Profiling Interface provides abilities to profile Bear code to optimize its
+performance.
+
+The profiler will start by profiling the run() method of bears because this is
+the part where bear writers will spend time on, as rest of the part like loading
+the files, collecting the settings, etc. are done by coala itself.
+
+.. note::
+
+    Enabling both profiler and debugger on a bear at the same time will result
+    in error.
+
+Bear writers will have the ability to directly dump the raw profile output
+either on current working directory or to a specified directory name, which can
+be further used for examination of profiler stats with the help of different
+modules like ``pstats`` or ``snakeviz``.
+
+Usage
+-----
+
+coala's Profiler accepts an additional parameter, a directory path where
+profiled files will be dumped, if ``True`` is provided profiled files will be
```

Comment:
ill avoid the next grammar lesson.

After `dumped` add a full stop.  Add two carraige returns.

Short simple intros are what people will read.  If the short intro is stuffed into a large paragraph, they will try to avoid it.

Then explain the `profile` arg options in a new paragraph.  And this usage section should not mention "True" , the title case Python constant, as there are multiple ways to enable profiling, and many do not require the Python keyword `True`.  Dumb it down.  Use prose.